Tyk 5.13.1 RNs.mdx (#2254)
TykTechnologies/tyk-docs: developer-support/release-notes/gateway.mdx

## 5.13 Release Notes 

### 5.13.1 Release Notes 

#### Release Date xx June 2026

#### Release Highlights

This patch implements several changes and fixes to improve the stability and correct the behavior of the Gateway.

The default content of application logs has been updated with RFC3339 format timestamp and consistent formatting with other components. This could be a **breaking change** for any user whose tooling relies on the specific content of application logs. A legacy mode is available, as described below.

For a comprehensive list of changes, please refer to the detailed [changelog](#Changelog-v5.13.1) below.

#### Breaking Changes

**Changes to application logs**

> This release makes [changes](#Changelog-v1.16.0) to the format of Tyk Gateway application logs so that output is consistent with other Tyk components and aligned with industry observability standards.

Reviewer: ```suggestion
This release makes changes to the format of Tyk Gateway application logs so that output is consistent with other Tyk components and aligned with industry observability standards.
```

<Note>
This is only a breaking change if you specifically rely on the current log format: for example, if your monitoring, parsing, or alerting tools are built around the existing timestamp format or the entire message.
</Note>

> **Mitigation:** A new `legacy` option has been added to the [`log_format`]() configuration. This setting preserves the format used prior to Tyk Gateway 5.8.15, with the historical timestamp format and lack of a field identifying for the originating API.

Reviewer: ```suggestion
**Mitigation:** A new `legacy` option has been added to the [`log_format`](/tyk-oss-gateway/configuration#log_format) configuration. This setting preserves the format used in Tyk Gateway 5.13.0 and earlier, with the historical timestamp format and lack of a field identifying for the originating API.
```

<a id="Changelog-v5.13.1" data-scroll-offset></a>

##### Fixed

<AccordionGroup>

<Accordion title='Fixed issue with Docker images incompatible with legacy registries and tools'>
{/* TT-17333 */}
Docker images published in v5.13.0 contained mixed layer compression — base image layers used zstd compression while Tyk application layers used gzip. This caused image copy and promotion failures with legacy Docker tooling, older registry versions, and some security scanners, with the error: `zstd compression is not officially supported for docker images`.

All image layers are now uniformly compressed using `gzip`, restoring compatibility with `skopeo`, `oc image mirror`, and other tools that do not support OCI `zstd` compression.
</Accordion>

<Accordion title='Standardized application log format'>
{/* TT-13727 */}
We have made changes to the format of Tyk Gateway application logs to make them consistent with logs generated by other Tyk components and aligned with industry observability standards.

- **RFC3339 timestamps:** Log entries now consistently use the industry-standard RFC3339 timestamp format (e.g. `2024-12-12T13:59:08Z`), previously a mix of timestamp formats was used.

> - **Log message field renamed:** The log message field is renamed from `msg` to `message`, for consistency with other Tyk components.

Reviewer: ```suggestion
```

Not actually implemented in 5.13.1

**Configuration**

The [`log_format`](/tyk-oss-gateway/configuration#log_format) configuration option now accepts three explicit values:
 
- `text`: plain-text logs in the new format introduced in this release (**new default**).
- `json`: JSON logs in the new format.

> - `legacy`: preserves the format used prior to Gateway 5.8.15, with the historical timestamp format and the `msg` field name.

Reviewer: ```suggestion
- `legacy`: preserves the format used in Gateway 5.13.0 and earlier, with the historical timestamp format.
```

> This is a **breaking change** for any user whose monitoring or log-parsing tools rely on the specific format of the Tyk Gateway application logs. If you require the previous timestamp format or the `msg` field name, you should set `log_format` to `legacy`.

Reviewer: ```suggestion
This is a **breaking change** for any user whose monitoring or log-parsing tools rely on the specific format of the Tyk Gateway application logs. If you require the previous timestamp format, you should set `log_format` to `legacy`.
```

Given the potential time difference between your upgrade and the release of this version, we recommend users verify the ongoing support of third-party dependencies they install, as their status may have changed since the release.

#### Deprecations

There are no deprecations in this release.

#### Upgrade instructions
<a id="upgrade-5.13.1"></a>

If you are upgrading to 5.13.1, please follow the detailed [upgrade instructions](#upgrading-tyk).

#### Downloads

- [Docker Image to pull](https://hub.docker.com/r/tykio/tyk-gateway/tags?page=&page_size=&ordering=&name=v5.13.1)
  - ```bash
    docker pull tykio/tyk-gateway:v5.13.1
    ```
- Helm charts
  - [tyk-charts](/developer-support/release-notes/helm-chart)

- [Source code tarball of Tyk Gateway v5.13.1](https://github.com/TykTechnologies/tyk/releases/tag/v5.13.1)

#### Changelog
<a id="Changelog-v5.13.1" data-scroll-offset></a>

##### Fixed

<AccordionGroup>

<Accordion title='Fixed issue with Docker images incompatible with legacy registries and tools'>
{/* TT-17333 */}

> 

Reviewer: ```suggestion
```
